### PR TITLE
Verify that COOP severed the relationship on the Opener's side, with the popup's initial Browsing context closed.

### PR DESCRIPTION
**Pull request by ParisMeuleman**
This adds am assertion to verify the windowProxy object is closed.

### CODE REVIEW

File: html/cross-origin-opener-policy/resources/postback.html [hunk 1]
```diff
@@ -5,6 +5,5 @@
 const bc = new BroadcastChannel(channelName);
 window.addEventListener("message", event => {
   bc.postMessage(event.data);
-  window.parent.close();
```
Comment (annevk) [inline html/cross-origin-opener-policy/resources/postback.html]: I think it's worth explaining why close is not called here.

File: html/cross-origin-opener-policy/resources/common.js [hunk 2]
```diff
@@ -12,12 +12,11 @@ function url_test(t, url, channelName, hasOpener, openerDOMAccess) {
     if (openerDOMAccess !== undefined) {
       assert_equals(payload.openerDOMAccess, openerDOMAccess, 'openerDOMAccess');
     }
+    assert_equals(w.closed, !hasOpener, 'Openee window proxy closed');
   });
 
   const w = window.open(url, channelName);
 
-  // w will be closed by its postback iframe. When out of process,
-  // window.close() does not work.
```
Comment (annevk) [inline html/cross-origin-opener-policy/resources/common.js]: You want to grep for "postback frame" and adjust similar callsites/comments I think. There's 2 or 3 in total iirc.

Comment (annevk) [reply html/cross-origin-opener-policy/resources/common.js]: (Also check the cross-origin-embedder-policy directory.)

File: html/cross-origin-opener-policy/resources/postback.html (hunk 1 above)
Comment (ParisMeuleman) [reply html/cross-origin-opener-policy/resources/postback.html]: Thanks Anne! I'm changing this to: the main test sending a message to the popup to trigger the closure during the cleanup. I'll push that new version to give you an idea of what I'm meaning here. I still need to update the other cases (as pointed out by your other comments).

I'm not entirely sure if the iframe is able to close its parent/top in all cases though, i.e. interpreting https://html.spec.whatwg.org/multipage/window-object.html#dom-window-close correctly. I think it is, but if not, I would add an extra hop where postback sends a message to the parent to trigger the closure...

Comment (annevk) [reply html/cross-origin-opener-policy/resources/postback.html]: There's a couple cases where neither gets to close it from script. This is mostly best effort for those running the tests manually.

**Comment by annevk** [comment]
Let me know when you think this is ready for review again as it's not immediately apparent. I think you still need to find some occurrences as I'm pretty sure that at least on master this also happens in html/cross-origin-embedder-policy.

**Comment by ParisMeuleman** [comment]
> Let me know when you think this is ready for review again as it's not immediately apparent. I think you still need to find some occurrences as I'm pretty sure that at least on master this also happens in html/cross-origin-embedder-policy.

I had a check on COEP cases, and I don't think we can have them close the popup:
  - the popup is openned with "noopener"
  - the popup navigates, which breaks the script-closable condition: "if it is a top-level browsing context whose session history contains only one Document."
This occurs in both none.https.html and requrie-corp.https.html.

I also expect coop-navigated-history-popup.https.html to be non-closable for the above reasons.

**Review by annevk — APPROVED** [review]
Assuming this doesn't affect how browsers fail/pass the tests, LGTM.

(Optional nits: "window proxy" -> "browsing context" (the WindowProxy object makes it observable), `if( ... )` -> `if (...)`.)

**Comment by ParisMeuleman** [comment]
> Assuming this doesn't affect how browsers fail/pass the tests, LGTM.
> 
> (Optional nits: "window proxy" -> "browsing context" (the WindowProxy object makes it observable), `if( ... )` -> `if (...)`.)

Thanks!, regarding the first nit, I did the update but I'm not sure it is accurate (or at least that my comments after the change are still accurate): Isn't the point of this that the bc (browsing context) is no longer observable by the opener, rather than the bc being closed?

**Comment by annevk** [comment]
The browsing context is closed and replaced by a new browsing context. Browsing context and WindowProxy are 1:1 and we're not changing that.

A spec-problem is that we might need something higher-level than a browsing context to keep track of transitions such as these. E.g., a "browsing session" or some such.

**Pull request merged**